components/director/pkg/resource/resource.go

```diff
@@ -1,77 +1,179 @@
-	// Vendor missing godoc
+	// Vendor type represents vendor resource.
 	Vendor Type = "vendor"
-	// Tombstone missing godoc
+	// Tombstone type represents tombstone resource.
 	Tombstone Type = "tombstone"
-	// IntegrationSystem missing godoc
+	// IntegrationSystem type represents integration system resource.
 	IntegrationSystem Type = "integrationSystem"
-	// SystemAuth missing godoc
+	// SystemAuth type represents system auth resource.
 	SystemAuth Type = "systemAuth"
-	// FetchRequest missing godoc
+	// FetchRequest type represents generic fetch request resource. This resource does not assume the referenced resource type of the FR.
 	FetchRequest Type = "fetchRequest"
-	// Specification missing godoc
+	// DocFetchRequest type represents document fetch request resource.
+	DocFetchRequest Type = "docFetchRequest"
+	// APISpecFetchRequest type represents API specification fetch request resource.
+	APISpecFetchRequest Type = "apiSpecFetchRequest"
+	// EventSpecFetchRequest type represents Event specification fetch request resource.
+	EventSpecFetchRequest Type = "eventSpecFetchRequest"
+	// Specification type represents generic specification resource. This resource does not assume the referenced resource type of the Spec.
 	Specification Type = "specification"
-	// Document missing godoc
+	// APISpecification type represents API specification resource.
+	APISpecification Type = "apiSpecification"
+	// EventSpecification type represents Event specification resource.
+	EventSpecification Type = "eventSpecification"
+	// Document type represents document resource.
 	Document Type = "document"
-	// BundleInstanceAuth missing godoc
+	// BundleInstanceAuth type represents bundle instance auth resource.
 	BundleInstanceAuth Type = "bundleInstanceAuth"
-	// API missing godoc
+	// API type represents api resource.
 	API Type = "api"
-	// EventDefinition missing godoc
+	// EventDefinition type represents event resource.
 	EventDefinition Type = "eventDefinition"
-	// AutomaticScenarioAssigment missing godoc
+	// AutomaticScenarioAssigment type represents ASA resource.
 	AutomaticScenarioAssigment Type = "automaticScenarioAssigment"
-	// Webhook missing godoc
+	// Webhook type represents generic webhook resource. This resource does not assume the referenced resource type of the Webhook.
 	Webhook Type = "webhook"
-	// Tenant missing godoc
+	// AppWebhook type represents application webhook resource.
+	AppWebhook Type = "appWebhook"
+	// RuntimeWebhook type represents runtime webhook resource.
+	RuntimeWebhook Type = "runtimeWebhook"
+	// Tenant type represents tenant resource.
 	Tenant Type = "tenant"
-	// TenantIndex missing godoc
-	TenantIndex Type = "tenantIndex"
-	// Schema missing godoc
+	// TenantAccess type represents tenant access resource.
+	TenantAccess Type = "tenantAccess"
+	// Schema type represents schema resource.
 	Schema Type = "schemaMigration"
 )
 
-// SQLOperation missing godoc
+var tenantAccessTable = map[Type]string{
```

Comment:
these postgre specific table mappings are placed in a pkg/resource/resource.go - it really does not make much sense in terms of wording atleast :)

Reply:
This is just a mapping between resource and it's tenant access table/view. Generally I see this package as a metadata stuff for each entity.

Can you elaborate why this does not make sense for you?

Even the SQL operations constants are here :D

Reply:
well, thats easy to answer :) - usually when you write code, the code should hint you that such a mapping is needed - the interface would do so by asking you to provide the mapping value. A static mapping does not hint you, probably you find out that exists the first time you run your code and you get an error. Best case scenario we get rid of the centralized mappings